src/common/scene-helpers.c

```diff
@@ -34,6 +44,285 @@ lab_wlr_scene_get_prev_node(struct wlr_scene_node *node)
 	return prev;
 }
 
+static double constrain(double lower, double in, double upper)
+{
```

Comment:
Tthis looks like it could be replaced by
```c
MAX(MIN(in, upper), lower)
```

Reply:
We could also add a new `CLAMP(in, lower, upper)` macro which just does the `MAX(MIN())` dance.

Reply:
Yes, that sounds better - will do that.